test_workspace/BinNat.bosatsu

```diff
@@ -97,11 +142,51 @@ def times2(b: BinNat) -> BinNat:
       #2(2(n + 1)) = 2((2n + 1) + 1)
       Even(Odd(n))
 
+def sub_BinNat(left: BinNat, right: BinNat) -> BinNat:
+  # invariant: left > right
```

Comment:
see below: invariant: left >= right, so the comment below is wrong, we can return Zero.